src/sources/kafka.rs

```diff
@@ -517,9 +562,13 @@ impl ClientContext for CustomContext {
 
 impl ConsumerContext for CustomContext {
     fn post_rebalance(&self, rebalance: &Rebalance) {
+        debug!("post_rebalance: {:#?}", rebalance);
         if matches!(rebalance, Rebalance::Revoke(_)) {
-            if let Some(finalizer) = self.finalizer.get() {
-                finalizer.flush();
+            if let Some(tx) = self.callback_channel.get() {
+                if let Err(error) = tx.send(KafkaCallback::PartitionRevoked) {
```

Comment:
Just to mention it here, since I assume this usage is what drove using an unbounded MPSC to begin with: the bounded MPSC sender has [`blocking_send`](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Sender.html#method.blocking_send) which could be used here.

Reply:
This code is in an awkward place: it's a sync function as required by `rdkafka`, so it can't use `tx.send(...).await`, but calling `blocking_send` here panics, because ultimately the MessageStream is being consumed within an async task (the top-level `select!`). It looks like  [`Handle::current().block_on(tx.send(...))`](https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html#method.block_on) would be a solution to this, are there any drawbacks or better solutions (other than keeping the channel unbounded)?

Reply:
Nevermind..I tried it and answered my own question. `block_on` also panics when used here :(

Reply:
Hmm, that's annoying. Let's split the difference: keep the unbounded channel, but always emit a DEBUG-level log if we're about to send a revocation message over the channel.

Essentially: I care less about the channel send failing, or that `post_balance` was called, than I do about being able to see (if need be) how many messages are going over that channel.